### ISSUE TITLE
The requirement of the trailing requires-clause in function-definition

### ISSUE THREAD

**Issue by xmh0511** [cplusplus/draft#4922]
Since in  p1971r0 where CA378 remove constrained non-template functions, the concrete rule is located in [dcl.decl.general] p4

> The optional requires-clause ([temp.pre]) in an init-declarator or member-declarator shall be present only if the declarator declares a templated function ([dcl.fct]). When present after a declarator, the requires-clause is called the trailing requires-clause. The trailing requires-clause introduces the constraint-expression that results from interpreting its constraint-logical-or-expression as a constraint-expression.

However, it is just about *requires-clause* in an **init-declarator or member-declarator**. Instead, the *function-definition* is a distinct grammar. How about the trailing requires-clause in a function-definition? Should it have had the same treatment with that in [dcl.decl.general] p4?

**Comment by xmh0511** [comment]
> [dcl.fct.def.general]/2 incorporates the [dcl.fct] rules.

requires-clause is not a part of a *declarator*. So, we say `void declarator;` or `declarator;` is well-formed, it didn't cover any requires-clause.

**Comment by opensdh** [comment]
Fair point: perhaps we should expand that rule, then, to avoid other duplication.

**Comment by xmh0511** [comment]
> Fair point: perhaps we should expand that rule, then, to avoid other duplication.

Agree. We should improve [dcl.decl.general] to cover all cases.
